src/components/TextAreaComponent.jsx

```diff
@@ -32,6 +38,7 @@ class TextArea extends Component {
     handleSubmit(event) {
         //TODO: perform tokenization here, save to state!!
         alert('A map was submitted: ' + this.state.text);
+        let tokenizer = new Tokenizer(this.state.text, literals);
```

Comment:
The reason I didn't make tokenizer a singleton is A. Elisa didn't (although she sooorta did) and B. Since we're not quitting the program each time like the in class examples we don't want to save the state of the Singleton. When you create the tokenizer you call tokenize() and the tokenizer actually saves state (i.e. the counter and tokens). We want new state and to call tokenize() again each time we click "submit".

Reply:
Mmm... alternatively we could have just one Tokenizer object and have a method that passes the string it has to tokenize. We could this method on every submit event, and then the tokenizer internally would, well, tokenize the string.

With this implementation, we just need to be careful to get the same instance from the AST nodes and not call the constructor. But anyway, even though there is a possibility of making an error, we can still make this work so maybe something we can go back and refactor _if we have time_. So resolving and approving your PR!